ktor-shared/ktor-serialization/ktor-serialization-gson/jvm/src/GsonConverter.kt

```diff
@@ -36,29 +52,84 @@ public class GsonConverter(private val gson: Gson = Gson()) : ContentConverter {
         return serializeNullable(contentType, charset, typeInfo, value)
     }
 
+    @Suppress("BlockingMethodInNonBlockingContext")
     override suspend fun serializeNullable(
         contentType: ContentType,
         charset: Charset,
         typeInfo: TypeInfo,
         value: Any?
     ): OutgoingContent {
-        return TextContent(gson.toJson(value), contentType.withCharsetIfNeeded(charset))
+        if (!streamRequestBody) {
+            return TextContent(gson.toJson(value), contentType.withCharsetIfNeeded(charset))
+        }
+        return OutputStreamContent(
+            {
+                val writer = this.writer(charset = charset)
+                // specific behavior for kotlinx.coroutines.flow.Flow : emit asynchronous values in Writer
+                if (typeInfo.type == Flow::class) {
+                    (value as Flow<*>).serializeJson(writer)
+                } else {
+                    // non flow content
+                    gson.toJson(value, writer)
+                }
+
+                // must flush manually
+                writer.flush()
+            },
+            contentType.withCharsetIfNeeded(charset)
+        )
     }
 
+    @OptIn(InternalAPI::class)
     override suspend fun deserialize(charset: Charset, typeInfo: TypeInfo, content: ByteReadChannel): Any? {
         if (gson.isExcluded(typeInfo.type)) {
             throw ExcludedTypeGsonException(typeInfo.type)
         }
 
         try {
             return withContext(Dispatchers.IO) {
+                // specific behavior for Sequence : collect it into a List
+                var isSequence = false
+                val resolvedTypeInfo = if (typeInfo.type == Sequence::class) {
+                    isSequence = true
+                    typeInfo.sequenceToListTypeInfo()
+                } else {
+                    typeInfo
+                }
                 val reader = content.toInputStream().reader(charset)
-                gson.fromJson(reader, typeInfo.reifiedType)
+                val decoded: Any? = gson.fromJson(reader, resolvedTypeInfo.reifiedType)
+
+                if (decoded != null && isSequence) {
+                    (decoded as List<*>).asSequence()
+                } else {
+                    decoded
+                }
             }
         } catch (deserializeFailure: JsonSyntaxException) {
             throw JsonConvertException("Illegal json parameter found", deserializeFailure)
         }
     }
+
+    private companion object {
+        private const val beginArrayCharCode = '['.code
```

Comment:
This for sure will break pretty print and maybe other configs.

Reply:
Thias is true, I missed it !

I can add the pretty print support for Flow serialization (with a new Unit test that verifies this use-case)